sdc/extensions/indexes/range_index_ext.py

```diff
@@ -0,0 +1,366 @@
+        ty_checker.raise_exc(step, 'number or none', 'step')
+
+    if not (isinstance(name, (types.NoneType, types.Omitted, types.StringLiteral, types.UnicodeType)) or name is None):
+        ty_checker.raise_exc(name, 'string or none', 'name')
+
+    if ((isinstance(start, (types.NoneType, types.Omitted)) or start is None)
+            and (isinstance(stop, (types.NoneType, types.Omitted)) or stop is None)
+            and (isinstance(step, (types.NoneType, types.Omitted)) or step is None)):
+        def pd_range_index_ctor_dummy_impl(
+                start=None, stop=None, step=None, dtype=None, copy=False, name=None, fastpath=None):
+            raise TypeError("RangeIndex(...) must be called with integers")
+
+        return pd_range_index_ctor_dummy_impl
+
+    def pd_range_index_ctor_impl(start=None, stop=None, step=None, dtype=None, copy=False, name=None, fastpath=None):
+
+        if not (dtype is None
+                or dtype == 'int64'
+                or dtype_is_np_int64):
+            raise TypeError("Invalid to pass a non-int64 dtype to RangeIndex")
+
+        _start = types.int64(start) if start is not None else types.int64(0)
+
+        if stop is None:
+            _start, _stop = types.int64(0), types.int64(start)
+        else:
+            _stop = types.int64(stop)
+
+        _step = types.int64(step) if step is not None else types.int64(1)
+        if _step == 0:
+            raise ValueError("Step must not be zero")
+
+        if _start == 0 and _stop == 0 and _step == 0:
+            raise TypeError("RangeIndex(...) must be called with integers")
+
+        return init_range_index(range(_start, _stop, _step), name)
+
+    return pd_range_index_ctor_impl
+
+
+@typeof_impl.register(pd.RangeIndex)
+def typeof_range_index(val, c):
+    is_named = val.name is not None
+    return RangeIndexType(is_named=is_named)
+
+
+@box(RangeIndexType)
+def box_range_index(typ, val, c):
+
+    mod_name = c.context.insert_const_string(c.builder.module, "pandas")
+    pd_class_obj = c.pyapi.import_module_noblock(mod_name)
+
+    range_index = numba.cgutils.create_struct_proxy(
+        typ)(c.context, c.builder, val)
+    range_index_data = numba.cgutils.create_struct_proxy(
+        RangeIndexDataType)(c.context, c.builder, range_index.data)
+
+    start = c.pyapi.from_native_value(types.int64, range_index_data.start)
+    stop = c.pyapi.from_native_value(types.int64, range_index_data.stop)
+    step = c.pyapi.from_native_value(types.int64, range_index_data.step)
+
+    # dtype and copy params are not stored so use default values
+    dtype = c.pyapi.make_none()
+    copy = c.pyapi.bool_from_bool(
+        c.context.get_constant(types.bool_, False)
+    )
+
+    if typ.is_named:
+        name = c.pyapi.from_native_value(types.unicode_type, range_index.name)
+    else:
+        name = c.pyapi.make_none()
+
+    res = c.pyapi.call_method(pd_class_obj, "RangeIndex", (start, stop, step, dtype, copy, name))
+
+    c.pyapi.decref(pd_class_obj)
```

Comment:
Shouldn't we also decrease reference for `start`, `stop` and `step` objects?
And for `dtype`, `copy` and `name`?

Reply:
Or, creating objects from native doesn't increment reference for it?

Reply:
Nice catch! Most likely yes, we need to decref for all python object we created using Python C API functions that return new references. It appears that we don't do this in box/unbox of Series and DFs which I used as a template too, but it's better to address that in a separate PR/task.